src/App.js

```diff
@@ -0,0 +1,22 @@
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import Header from './component/hearder';
+import ProductList from './component/productList';
+import ProductDetail from './component/productDetail';
+import './component/componets.css';
+
+function App() {
+  return (
+    <div className="App">
+      <Router>
```

Comment:
You successfully created a  single page application (SPA) using `react-router-dom` to serve each page without reloading 🙌🔥